pcsx2/gui/FrameForGS.cpp

```diff
@@ -155,11 +155,22 @@ void GSPanel::DoResize()
 
 	double clientAr = (double)client.GetWidth()/(double)client.GetHeight();
 
+	extern AspectRatioType iniAR;
+	extern bool switchAR;
 	double targetAr = clientAr;
-	if( g_Conf->GSWindow.AspectRatio == AspectRatio_4_3 )
```

Comment:
I'm wondering whether you could just extend the if so it's
`if( g_Conf->GSWindow.AspectRatio == AspectRatio_4_3 || (g_Conf->GSWindow.IsToggleAspectRatioSwitch && FMVswitch))`

It may simplify your current solution (of course, I may have overlooked something too).